lib/improver/cli/nbhood_land_and_sea.py

```diff
@@ -243,7 +155,7 @@ def process(cube, mask, radius=None, radii_by_lead_time=None, weights=None,
         sea_only = landmask.copy(data=landmask.data.astype(int))
 
     else:
-        if weights is None:
+        if weights is not None:
```

Comment:
This was broken before.

Reply:
We should probably get rid of warnings that are so ignored we don't even notice they're being wrongly triggered ... @TomekTrzeciak

Reply:
I agree. In this case, either error or kill the warning. I think we probably want to error, because we shouldn't be passing in the weights cube if it's a no-op - could be misleading.

Reply:
Changed into a TypeError, or do you think something different?